xds/src/main/java/com/linecorp/armeria/xds/XdsConverterUtil.java

```diff
@@ -52,13 +78,21 @@ static List<Endpoint> convertEndpoints(ClusterLoadAssignment clusterLoadAssignme
                             } else {
                                 return Endpoint.of(socketAddress.getAddress(), socketAddress.getPortValue());
                             }
-                        })).collect(Collectors.toList());
+                        })).collect(toImmutableList());
     }
 
-    static List<Endpoint> convertEndpoints(Collection<ClusterLoadAssignment> clusterLoadAssignments) {
-        return clusterLoadAssignments.stream()
-                                     .flatMap(cla -> convertEndpoints(cla).stream())
-                                     .collect(Collectors.toList());
+    private static boolean containsExactlyInAnyOrder(Struct metadata1, Struct metadata2) {
+        if (metadata1.getFieldsCount() != metadata2.getFieldsCount()) {
+            return false;
+        }
+        final Map<String, Value> metadata2Map = metadata2.getFieldsMap();
+        for (Entry<String, Value> entry : metadata1.getFieldsMap().entrySet()) {
+            final Value value = metadata2Map.get(entry.getKey());
+            if (value == null || !value.equals(entry.getValue())) {
```

Comment:
Note: List values are actually handled differently in envoy, but I think it over-complicates the implementation.
I think the current approach is fine 👍 

ref: https://github.com/envoyproxy/envoy/blob/main/source/extensions/load_balancing_policies/subset/subset_lb.cc#L669

Reply:
Will revisit this later when I need to. Thanks!